filters/fadein/fadein.go

```diff
@@ -192,11 +193,20 @@ func (endpointCreated) CreateFilter(args []interface{}) (filters.Filter, error)
 func (endpointCreated) Request(filters.FilterContext)  {}
 func (endpointCreated) Response(filters.FilterContext) {}
 
+type PostProcessorOptions struct {
+	EndpointRegistry *routing.EndpointRegistry
+}
+
 // NewPostProcessor creates post-processor for maintaining the detection time of LB endpoints with fade-in
 // behavior.
-func NewPostProcessor() routing.PostProcessor {
+func NewPostProcessor(options PostProcessorOptions) routing.PostProcessor {
+	if options.EndpointRegistry == nil {
+		options.EndpointRegistry = routing.NewEndpointRegistry(routing.RegistryOptions{})
+	}
```

Comment:
This makes no sense as this registry instance will no be visible outside of this postprocessor.
The code should call GetMetrics only when registry is not nil instead.